cluster-autoscaler/cloudprovider/hetzner/hetzner_node_group.go

```diff
@@ -91,7 +92,8 @@ func (n *hetznerNodeGroup) IncreaseSize(delta int) error {
 		return fmt.Errorf("delta must be positive, have: %d", delta)
 	}
 
-	targetSize := n.targetSize + delta
+	expectedDelta := delta
```

Comment:
I struggle to understand the different between `n.targetSize`, `targetSize`, `delta` and `expectedDelta`. Can the naming be improved, or commented?

Reply:
- `n.targetSize` is the count of servers that currently exists. Its exposed through `n.TargetSize()` and has additional docs there.
- `targetSize` is the count of servers that should exist after the current call. Its only used to verify against the configured limit. I have renamed this to `desiredTargetSize`
- `delta` is the count of additional nodes that should be added. This is "given" from the interface we implement.
- `expectedDelta` is the count of additional nodes we actually added. `delta` minus any failures that occured. I have renamed this to `actualDelta`.

Realistically this could be all removed and we can just get the count of nodes from the servers cache when we need it. This code is older than the cache, without a cache it makes sense to keep the exact number of servers around for these checks. Ill open another PR to remove it